### PR TITLE
Helm: added HttpListenPort in distributor-svc-headless

### LINKED ISSUE #4392: Helm: Lost http port in distributor headless service
#### Describe the bug

Lost http port in distributor headless service. The default port is used in nginx-configmap gateway.

#### To Reproduce

Generate template grafana/mimir-distributed with values.yaml file, where set `gateway.enabledNonEnterprise: true`
Check distributor/distributor-svc-headless.yaml and gateway/nginx-configmap.yaml.

<details>
  <summary>distributor/distributor-svc-headless.yaml</summary>

```yaml
# Source: mimir-distributed/templates/distributor/distributor-svc-headless.yaml
apiVersion: v1
kind: Service
metadata:
  name: mimir-distributor-headless
  labels:
    helm.sh/chart: mimir-distributed-4.2.0
    app.kubernetes.io/name: mimir
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/component: distributor
    app.kubernetes.io/part-of: memberlist
    app.kubernetes.io/version: "2.6.0"
    app.kubernetes.io/managed-by: Helm
    prometheus.io/service-monitor: "false"
  annotations:
    {}
  namespace: "monitoring"
spec:
  type: ClusterIP
  clusterIP: None
  ports:
    - port: 9095
      protocol: TCP
      name: grpc
      targetPort: grpc
  selector:
    app.kubernetes.io/name: mimir
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/component: distributor
```
</details>

<details>
  <summary>gateway/nginx-configmap.yaml</summary>

```yaml
# Source: mimir-distributed/templates/gateway/nginx-configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: mimir-gateway-nginx
  labels:
    helm.sh/chart: mimir-distributed-4.2.0
    app.kubernetes.io/name: mimir
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/component: gateway-nginx
    app.kubernetes.io/version: "2.6.0"
    app.kubernetes.io/managed-by: Helm
  namespace: "monitoring"
data:
  nginx.conf: |
    worker_processes  5;  ## Default: 1
    error_log  /dev/stderr;
    pid        /tmp/nginx.pid;
    worker_rlimit_nofile 8192;
    events {
      worker_connections  4096;  ## Default: 1024
    }
    http {
      client_body_temp_path /tmp/client_temp;
      proxy_temp_path       /tmp/proxy_temp_path;
      fastcgi_temp_path     /tmp/fastcgi_temp;
      uwsgi_temp_path       /tmp/uwsgi_temp;
      scgi_temp_path        /tmp/scgi_temp;
      default_type application/octet-stream;
      log_format   main '$remote_addr - $remote_user [$time_local]  $status '
            '"$request" $body_bytes_sent "$http_referer" '
            '"$http_user_agent" "$http_x_forwarded_for"';
      map $status $loggable {
        ~^[23]  0;
        default 1;
      }
      access_log   /dev/stderr  main  if=$loggable;
      sendfile     on;
      tcp_nopush   on;
      resolver kube-dns.kube-system.svc.cluster.local;
      # Ensure that X-Scope-OrgID is always present, default to the no_auth_tenant for backwards compatibility when multi-tenancy was turned off.
      map $http_x_scope_orgid $ensured_x_scope_orgid {
        default $http_x_scope_orgid;
        "" "anonymous";
      }
      server {
        listen 8080;
        location = / {
          return 200 'OK';
          auth_basic off;
        }
        location = /ready {
          return 200 'OK';
          auth_basic off;
        }
        proxy_set_header X-Scope-OrgID $ensured_x_scope_orgid;
        # Distributor endpoints
        location /distributor {
          proxy_pass      http://mimir-distributor-headless.monitoring.svc.cluster.local:8080$request_uri;
        }
        location = /api/v1/push {
          proxy_pass      http://mimir-distributor-headless.monitoring.svc.cluster.local:8080$request_uri;
        }
        location /otlp/v1/metrics {
          proxy_pass      http://mimir-distributor-headless.monitoring.svc.cluster.local:8080$request_uri;
        }
        # Alertmanager endpoints
        location /alertmanager {
          proxy_pass      http://mimir-alertmanager-headless.monitoring.svc.cluster.local:8080$request_uri;
        }
        location = /multitenant_alertmanager/status {
          proxy_pass      http://mimir-alertmanager-headless.monitoring.svc.cluster.local:8080$request_uri;
        }
        location = /api/v1/alerts {
          proxy_pass      http://mimir-alertmanager-headless.monitoring.svc.cluster.local:8080$request_uri;
        }
        # Ruler endpoints
        location /prometheus/config/v1/rules {
          proxy_pass      http://mimir-ruler.monitoring.svc.cluster.local:8080$request_uri;
        }
        location /prometheus/api/v1/rules {
          proxy_pass      http://mimir-ruler.monitoring.svc.cluster.local:8080$request_uri;
        }
        location /prometheus/api/v1/alerts {
          proxy_pass      http://mimir-ruler.monitoring.svc.cluster.local:8080$request_uri;
        }
        location = /ruler/ring {
          proxy_pass      http://mimir-ruler.monitoring.svc.cluster.local:8080$request_uri;
        }
        # Rest of /prometheus goes to the query frontend
        location /prometheus {
          proxy_pass      http://mimir-query-frontend.monitoring.svc.cluster.local:8080$request_uri;
        }
        # Buildinfo endpoint can go to any component
        location = /api/v1/status/buildinfo {
          proxy_pass      http://mimir-query-frontend.monitoring.svc.cluster.local:8080$request_uri;
        }
        # Compactor endpoint for uploading blocks
        location /api/v1/upload/block/ {
          proxy_pass      http://mimir-compactor.monitoring.svc.cluster.local:8080$request_uri;
        }
      }
    }

```

</details>

#### Expected behavior

distributor/distributor-svc-headless.yaml must have an http port, because it is used in configmap gateway

#### Environment
None

#### Additional Context

None

### PR DESCRIPTION
**Pull request by sentoz**
#### What this PR does

Added http port for distributor-svc-headless.

#### Which issue(s) this PR fixes or relates to

Fixes #4392 

#### Checklist

- [x] Tests updated
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 * [ENHANCEMENT] Support autoscaling/v2 HorizontalPodAutoscaler for nginx autoscaling starting with Kubernetes 1.23. #4285
 * [BUGFIX] Allow override of Kubernetes version for nginx HPA. #4299
 * [BUGFIX] Do not generate query-frontend-headless service if query scheduler is enabled. Fixes parity with jsonnet. #4353
+* [BUGFIX] Lost http port in distributor headless service. #4392
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:38]:
```suggestion
* [BUGFIX] Add http port in distributor headless service. Fixes parity with jsonnet. #4392
```

File: operations/helm/charts/mimir-distributed/templates/distributor/distributor-svc-headless.yaml [hunk 2]
```diff
@@ -15,6 +15,10 @@ spec:
   type: ClusterIP
   clusterIP: None
   ports:
+    - port: {{ include "mimir.serverHttpListenPort" .}}
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/templates/distributor/distributor-svc-headless.yaml:18]: The one concern I have is whether metamonitoring will start scraping this automatically, resulting in duplicated distributor metrics. Needs a test - currently manual.

Comment (sentoz) [reply operations/helm/charts/mimir-distributed/templates/distributor/distributor-svc-headless.yaml:18]: The question is fair.
I have one servicemonitor for the entire mimir instance, I did not notice any duplication in the metrics. I will study more carefully.

Comment (sentoz) [reply operations/helm/charts/mimir-distributed/templates/distributor/distributor-svc-headless.yaml:18]: And you are not worried about a possible similar problem for other mimir components, such as query-frontend. There is also an http port in the service and in the headless service

Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/templates/distributor/distributor-svc-headless.yaml:18]: we have the `prometheus.io/service-monitor: "false"` annotation on all headless services, this shouldn't be a problem IIRC

Comment (sentoz) [reply operations/helm/charts/mimir-distributed/templates/distributor/distributor-svc-headless.yaml:18]: @krajorama @dimitarvdimitrov In that case, there are no more problems to accept the PR?

Comment (krajorama) [reply operations/helm/charts/mimir-distributed/templates/distributor/distributor-svc-headless.yaml:18]: I think not, please update the changelog with the suggestion and for latest main. And we can merge.

**Review by krajorama — APPROVED** [review]
Approved with comment. Thank you for the contribution.

**Comment by sentoz** [comment]
Sorry, I took the edits from main via forcepush(

**Pull request merged**